[npm]Publish updated npm packages for @celo/walletkit and @celo/celocli

ashishb opened the pull request:
### Description

Publish updated npm packages for @celo/walletkit and @celo/celocli

Few two more bugs

1. Export `bin/build-sdk.js` from the walletkit
2. For ocliff, commands are in`"commands": "./lib/src/commands",`
3. Generate ocliff manifest before we publish

### Tested

```
$ docker run -v $PWD/packages/walletkit:/tmp/npm_package -it --entrypoint bash node:8
$ npm install /tmp/npm_package # Inside docker
```

```
$ docker run -v $PWD/packages/cli:/tmp/npm_package -it --entrypoint bash node:8
$ npm install /tmp/npm_package # Inside docker
```

mcortesi commented on packages/walletkit/package.json:
```diff
@@ -59,6 +59,7 @@
     "index.ts",
-    "lib/**/*"
+    "lib/**/*",
+    "bin/build-sdk.js"
```
why do we need this?

ashishb replied:
The build step of celocli fails otherwise.


```
$ docker run -v $PWD/packages/cli:/tmp/npm_package -it --entrypoint bash node:8
root@52b17790feb9:/# npm install /tmp/npm_package/
npm WARN deprecated left-pad@1.3.0: use String.prototype.padStart()
npm WARN deprecated fs-promise@2.0.3: Use mz or fs-extra^3.0 with Promise Support
npm WARN deprecated tar.gz@1.0.7: ⚠️  WARNING ⚠️ tar.gz module has been deprecated and your application is vulnerable. Please use tar module instead: https://npmjs.com/tar

> scrypt@6.0.3 preinstall /node_modules/scrypt
> node node-scrypt-preinstall.js


> scrypt@6.0.3 preinstall /node_modules/ganache-core/node_modules/scrypt
> node node-scrypt-preinstall.js

npm WARN enoent ENOENT: no such file or directory, open '/package.json'
npm WARN bitcore-mnemonic@1.7.0 requires a peer of bitcore-lib@^0.16.0 but none is installed. You must install peer dependencies yourself.
npm WARN !invalid#1 No description
npm WARN !invalid#1 No repository field.
npm WARN !invalid#1 No README data
npm WARN !invalid#1 No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/ganache-core/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! path /node_modules/@celo/walletkit/bin/build-sdk.js
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod
npm ERR! enoent ENOENT: no such file or directory, chmod '/node_modules/@celo/walletkit/bin/build-sdk.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-08-13T22_27_25_140Z-debug.log
```

ashishb replied:
As discussed, I moved `build` step of `walletkit` to `prepublishOnly` for now.

ashishb committed:
[npm]Publish updated npm packages for @celo/walletkit and @celo/celocli

Publish updated npm packages for @celo/walletkit and @celo/celocli

```
$ docker run -v $PWD/packages/walletkit:/tmp/npm_package -it --entrypoint bash node:8
$ npm install /tmp/npm_package # Inside docker
```

```
$ docker run -v $PWD/packages/cli:/tmp/npm_package -it --entrypoint bash node:8
$ npm install /tmp/npm_package # Inside docker
```

mcortesi commented on packages/cli/package.json:
```diff
@@ -21,13 +21,15 @@
   },
   "scripts": {
-    "build": "rm -rf lib && tsc -b",
+    "build": "rm -rf lib && yarn run prepack",
```
dont run prepack on build

ashishb replied:
I was reviving the old codebase. Removed.

mcortesi commented on packages/cli/package.json:
```diff
@@ -21,13 +21,15 @@
+    "build": "rm -rf lib && yarn run prepack",
     "docs": "yarn oclif-dev readme --multi --dir=../docs/command-line-interface && yarn prettier ../docs/command-line-interface/*.md --write",
     "lint": "tslint -c tslint.json --project tsconfig.json",
+    "prepack": "tsc -b && oclif-dev manifest && oclif-dev readme",
```
remove.
and make `prepublishOnly: "yarn build && oclif-dev-manifed"`

ashishb replied:
I was reviving the old codebase. Removed.

mcortesi commented on packages/walletkit/package.json:
```diff
@@ -9,10 +9,11 @@
     "build": "yarn run build:for-env alfajores",
+    "clean": "rm -rf lib .artifacts contracts types lib",
+    "lint": "tslint -c tslint.json --project tsconfig.json",
+    "prepublishOnly": "yarn run prebuild && yarn run build",
```
no need to do prebuild.
whenever you do `yarn run XXX`... all `preXXX` and `posXXX` are run automatically

ashishb committed:
Merge branch 'master' into ashishb/new_npm_packages

ashishb committed:
Merge branch 'master' into ashishb/new_npm_packages

Commit:
Merge refs/heads/master into ashishb/new_npm_packages

Commit:
Merge refs/heads/master into ashishb/new_npm_packages

Commit:
Merge refs/heads/master into ashishb/new_npm_packages

Commit:
Merge refs/heads/master into ashishb/new_npm_packages

Commit:
Merge refs/heads/master into ashishb/new_npm_packages
